### DISCUSSION TITLE
Storage Copy/Move Across Buckets Only Manipulates Source Bucket

### DISCUSSION THREAD

**Discussion by shepblaster** [supabase/supabase#26624] [Questions] [answered]
Using Storage with the Javascript Client Library. Trying to copy a file from one bucket to another. 

Following the instructions here: 
https://supabase.com/docs/guides/storage/management/copy-move-objects

Directed to (arguments changed in my code for my use case):
await supabase.storage.from('avatars').copy('public/avatar1.png', 'private/avatar2.png', {
  destinationBucket: 'avatars2',
})

Result:
The file to be copied is copied, but placed in the same bucket as it started in. destinationBucket is unchanged

I am using a service level key to make these changes. Have tried with no policies on both buckets, and policies permitting the service role to take any action in both the from and destinationBucket

Any help would be greatly appreciated!

**Comment by GaryAustin1** [comment, marked as answer]
It was released in storage-js...  https://github.com/supabase/storage-js/pull/197
But that version of storage-js has not been merged into the latest supabase-js yet.

**Reply by shepblaster** [reply]
Understood! Thank you, Gary. 

Kind of a noob at understanding the github processes for a project like this. 

I go [here](https://github.com/supabase/storage-js/commits/main/) and see that the commit is kind of stalled out

Any way for the public to help this get pushed through, or do we just wait for fenos to handle it? 
![checks](https://github.com/supabase/supabase/assets/113643009/9d87d885-6726-4114-960c-59fc9dcb7230)

**Reply by GaryAustin1** [reply]
I would also comment on his post. 
Maybe generate an issue in supabase-js asking where it is?   
Seems bad to show it in the docs with JS and it is just sitting waiting for release.

**Reply by shepblaster** [reply]
Will do. Thank you for the help.
